doc/AUTOMATIC_BOOT_ASSESSMENT.md

```diff
@@ -0,0 +1,197 @@
+# Automatic Boot Assessment
+
+For unattended systems such as servers, IoT/embedded devices or appliances an
+automatic logic of reverting back to previous versions of the OS or kernel in
+case system boots consistently fail is important. systemd provides support for
+this in various of its components. When used together they provide a complete
+solution on UEFI systems, built as add-on to the [Boot Loader
+Specification](https://github.com/systemd/systemd/blob/master/doc/BOOT_LOADER_SPECIFICATION.md). However,
+the different components may also be used independently, and in combination
+with other software, to implement similar schemes, for example with other boot
+loaders or for non-UEFI systems. Here's a brief overview of the complete set of
+components:
+
+* The
+  [`systemd-boot(7)`](https://www.freedesktop.org/software/systemd/man/systemd-boot.html)
+  boot loader optionally maintains a per-boot-loader-entry counter that is
+  decreased by one on each attempt to boot the entry, priorizing entries that
+  have non-zero counters over those which already reached a counter of zero
+  when choosing the entry to boot.
+
+* The
+  [`systemd-boot-complete.service(8)`](https://www.freedesktop.org/software/systemd/man/systemd-boot-complete.service.html)
+  service automatically marks a boot loader entry for which boot counting as
+  mentioned above is enabled, as "good" when a boot has been determined to be
+  successful, thus turning off boot counting for it again.
+
```

Comment:
One comma is missing: "marks a boot loader **entry,** for which boot counting as mentioned above is enabled, as "good" when a boot has been determined to be successful".